service/system_information.go

```diff
@@ -0,0 +1,330 @@
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get storage pools on %q: %w", s.ClusterName, err)
+	}
+
+	pool, ok := pools[DefaultZFSPool]
+	if ok {
+		poolCopy := pool
+		s.existingLocalPool = &poolCopy
+	}
+
+	pool, ok = pools[DefaultCephPool]
+	if ok {
+		poolCopy := pool
+		s.existingRemotePool = &poolCopy
+	}
+
+	pool, ok = pools[DefaultCephFSPool]
+	if ok {
+		poolCopy := pool
+		s.existingRemoteFSPool = &poolCopy
+	}
+
+	if len(s.ExistingServices[types.MicroCeph]) > 0 {
+		microceph := sh.Services[types.MicroCeph].(*CephService)
+
+		if localSystem {
+			s.CephConfig, err = microceph.ClusterConfig(ctx, "", "")
+		} else {
+			s.CephConfig, err = microceph.ClusterConfig(ctx, s.ClusterAddress, s.AuthSecret)
+		}
+
+		if err != nil && !api.StatusErrorCheck(err, http.StatusServiceUnavailable) {
+			return nil, fmt.Errorf("Failed to get ceph configuration on %q: %w", s.ClusterName, err)
+		}
+	}
+
+	if localSystem {
+		s.LXDLocalConfig, s.LXDConfig, err = lxd.GetConfig(ctx, s.ServiceClustered(types.LXD), s.ClusterName, "", "")
+	} else {
+		s.LXDLocalConfig, s.LXDConfig, err = lxd.GetConfig(ctx, s.ServiceClustered(types.LXD), s.ClusterName, s.ClusterAddress, s.AuthSecret)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get LXD configuration on %q: %w", s.ClusterName, err)
+	}
+
+	return s, nil
+}
+
+// GetExistingClusters checks against the services reachable by the specified ServerInfo,
+// and returns a map of cluster members for each service supported by the Handler.
+// If a service is not clustered, its map will be nil.
+func (sh *Handler) GetExistingClusters(ctx context.Context, connectInfo mdns.ServerInfo) (map[types.ServiceType]map[string]string, error) {
+	localSystem := sh.Name == connectInfo.Name
+	var err error
+	existingServices := map[types.ServiceType]map[string]string{}
+	for service := range sh.Services {
+		var existingCluster map[string]string
+		if localSystem {
+			existingCluster, err = sh.Services[service].ClusterMembers(ctx)
+		} else {
+			existingCluster, err = sh.Services[service].RemoteClusterMembers(ctx, connectInfo.AuthSecret, connectInfo.Address)
+		}
+
+		if err != nil && !api.StatusErrorCheck(err, http.StatusServiceUnavailable) {
+			return nil, fmt.Errorf("Failed to reach %s on system %q: %w", service, connectInfo.Name, err)
+		}
+
+		// If a service isn't clustered, this loop will be skipped.
+
+		for k, v := range existingCluster {
+			if existingServices[service] == nil {
+				existingServices[service] = map[string]string{}
+			}
+
+			host, _, err := net.SplitHostPort(v)
+			if err != nil {
+				return nil, err
+			}
+
+			existingServices[service][k] = host
+		}
+	}
+
+	return existingServices, nil
+}
+
+// SupportsLocalPool checks if the SystemInformation supports a MicroCloud configured local storage pool.
+// Additionally returns whether such a pool already exists.
+func (s *SystemInformation) SupportsLocalPool() (hasPool bool, supportsPool bool) {
```

Comment:
What is the difference between `hasPool` and `supportsPool`? 
It's hard to tell based on the three return values.

Reply:
`hasPool` indicates that the storage pool already exists with a configuration that MicroCloud can work with (so for example a `zfs` storage pool named `local` already exists)

`supportsPool` indicates that LXD is incompatible with the storage pool in the way MicroCloud would set it up (for example there is a `dir` storage pool named `local` so we can't create a `zfs` storage pool named `local`)

Reply:
Ah I see `existingLocalPool` gets set at another place if there is a `local` entry in the respective map.

Thanks clear now.